### ISSUE TITLE
Install fails on Linux (error: Could not compile `serde_derive`.)

### ISSUE THREAD

**Issue by x4121** [alexcrichton/cargo-vendor#32]
Running `cargo install cargo-vendor` (even with an empty registry) results in:

```
   ...
   Compiling curl v0.4.6
   Compiling serde_derive v0.9.15
error[E0523]: found two different crates with name `std` that are not distinguished by differing `-C metadata`. This will result in symbol conflicts between the two.
  --> $my_home/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_derive-0.9.15/src/lib.rs:13:1
   |
13 | extern crate proc_macro;
   | ^^^^^^^^^^^^^^^^^^^^^^^^

error: Could not compile `serde_derive`.
Build failed, waiting for other jobs to finish...
error: failed to compile `cargo-vendor v0.1.7`, intermediate artifacts can be found at `/tmp/cargo-install.Y9F5h9aJKgW6`
```

Same happens when building `master` or `0.1.7` with `cargo build`, `0.1.6` works fine.

## Env
Ubuntu 17.04
rustc 1.16.0
cargo-0.17.0 (built 2017-03-24)

## To reproduce
```
docker run --rm -it ubuntu:zesty bash
$ apt update && apt install -y cargo pkg-config libssl-dev libssh2-1-dev
$ cargo install cargo-vendor
```

**Comment by alexcrichton** [comment]
Thanks for the report! I think this is a bug in how rust/cargo are packaged for ubuntu though? This doesn't reproduce locally for me with the official toolchains.

**Comment by x4121** [comment]
OK, can confirm this. Built rust/cargo from source and the error doesn't occur.
Maybe it's something introduced/fixed between rustc 1.16.0 and 1.19.0 or cargo 0.17.0 and 0.20.0 :slightly_frowning_face:
